shellwhat/test_exercise.py

```diff
@@ -22,29 +23,37 @@ def test_exercise(
     """
     """
 
-    state = State(
-        student_code=student_code,
-        solution_code=solution_code,
-        pre_exercise_code=pre_exercise_code,
-        student_conn=student_conn,
-        solution_conn=solution_conn,
-        student_result=student_result,
-        solution_result=solution_result,
-        reporter=Reporter(errors=error),
-        force_diagnose=force_diagnose,
-    )
-
-    State.root_state = state
-    SCT_CTX["Ex"].root_state = state
+    reporter = Reporter(errors=error)
 
-    try:
-        exec(sct, SCT_CTX)
-    except TestFail as tf:
-        result = tf.payload
-        result["student_code"] = student_code
-        return result
-
-    result = state.reporter.build_final_payload()
-    result["student_code"] = student_code
+    def add_student_code(data):
+        data["student_code"] = student_code
+        return data
 
-    return result
+    try:
```

Comment:
State needs to be in try except? (See sqlwhat?)

Reply:
In python the code is parsed when State is created, because we want to be able to do it similarly in other languages the State creation should also be in the try, except.